apps/web-mzima-client/src/app/post/media-uploader/media-uploader.component.ts

```diff
@@ -0,0 +1,290 @@
+            const uploadObservable: Observable<any> = this.mediaService
+              .uploadFileProgress(aMediaFile.file!, '')
+              .pipe(
+                tap((uploadEvent) => {
+                  // if (uploadEvent.type === HttpEventType.UploadProgress) {
+                  //   const progressEvent: HttpProgressEvent = uploadEvent as HttpProgressEvent;
+                  //   const percentDone = progressEvent.total
+                  //     ? progressEvent.loaded / progressEvent.total
+                  //     : 0;
+
+                  //   // if (this.progressCallback)
+                  //   //   this.progressCallback(percentDone);
+                  // }
+                  // else
+                  if (uploadEvent.type === HttpEventType.Response) {
+                    this.updateMediaFileById(
+                      aMediaFile.generatedId,
+                      uploadEvent.body,
+                      (mediaFile, resultBody) => {
+                        mediaFile.status = 'uploaded';
+                        mediaFile.value = resultBody.result.id;
+                        return mediaFile;
+                      },
+                    );
+                    // if (this.progressCallback)
+                    //     this.progressCallback(100);
+                    setTimeout(
+                      (mediaFile: MediaFile) => {
+                        this.updateMediaFileById(
+                          mediaFile.generatedId,
+                          uploadEvent.body,
+                          (theMediaFile) => {
+                            theMediaFile.status = 'ready';
+                            return theMediaFile;
+                          },
+                        );
+                      },
+                      3000,
+                      aMediaFile,
+                    );
+                  }
+                }),
+                last(),
+                catchError((error: HttpErrorResponse) => {
+                  this.updateMediaFileById(aMediaFile.generatedId, null, (mediaFile) => {
+                    mediaFile.status = 'error';
+                    return mediaFile;
+                  });
+                  return throwError(() => new Error(error.statusText));
+                }),
+              );
+            this.uploads.set(aMediaFile.generatedId, uploadObservable);
+          });
+        forkJoin(Array.from(this.uploads.values())).subscribe((results) => {
+          for (const result of results) {
+            const filename = getFileNameFromUrl(result.body.result.original_file_url);
+            this.updateMediaFileByNameAndSize(
+              filename,
+              result.body.result.original_file_size,
+              (mediaFile) => {
+                mediaFile.value = result.body.result.id;
+                return mediaFile;
+              },
+            );
+          }
+          this.onChange(this.mediaFiles);
+          inputElement.value = '';
+          console.log(results);
```

Comment:
@ushahidlee 👋    pointing at this console.log() just in case you didn't have the intention to leave it